File: src-tauri/src/models/clipboard_item.rs
use serde::Serialize;

#[derive(Debug, Serialize, Clone)]
pub enum ClipboardKind {
    text,
    link,
    code,
    image,
    file,
}

#[derive(Clone, Serialize)]
pub struct ClipboardItem {
    pub id: String,
    pub kind: ClipboardKind,
    pub content: String,
    pub file_path: Option<String>,
    pub created_at: i64,
    pub expires_at: i64,
}

File: src/types/clipboardItem.ts
import { z } from "zod";

export const ClipboardItemSchema = z.object({
  id: z.string(),
  kind: z.enum(["text", "image", "code", "file", "link", "location"]),
  content: z.string(),
  filepath: z.string().optional(),
  created_at: z.number(), 
  expires_at: z.number().optional(),
});


export type ClipboardItem = z.infer<typeof ClipboardItemSchema>;
export const ClipboardItemArraySchema = z.array(ClipboardItemSchema);